src/azul/indexer/document.py

```diff
@@ -1541,11 +1550,33 @@ def field_types(cls, field_types: FieldTypes) -> FieldTypes:
     def to_json(self) -> JSON:
         return dict(super().to_json(),
                     replica_type=self.replica_type,
-                    hub_ids=self.hub_ids)
+                    hub_ids=sorted(self.hub_ids))
 
     @property
     def op_type(self) -> OpType:
-        return OpType.create
+        if self.version_type is VersionType.create_only:
+            return OpType.create
+        elif self.version_type is VersionType.none:
+            return OpType.update
+        else:
+            assert False, self.version_type
+
+    def _body(self, field_types: FieldTypes) -> JSON:
+        if self.op_type is OpType.update:
+            return {
+                'script': {
+                    'source': '''
+                    ctx._source.hub_ids.addAll(params.hub_ids);
```

Comment:
```suggestion
                    Stream stream = Stream.concat(ctx._source.hub_ids.stream(), params.hub_ids.stream());
                    ctx._source.hub_ids = stream.sorted().distinct().collect(Collectors.toList());
```

https://stackoverflow.com/questions/43806467/java-streams-how-to-do-an-efficient-distinct-and-sort